### ISSUE TITLE
Add an endpoint/parameter that redirects to a random instance

### ISSUE THREAD

**Issue by bohwaz** [PrivateBin/Directory#33]
Basically the same as the /api endpoint, with the same available parameters, but will redirect to one of the instances matching the criterias, randomly.

Eg:

```
curl -v https://privatebin.info/directory/api?https=true&redirect=true
...
Location: https://privatebin.randomserver.net/
```

Or an endpoint:

```
curl -v https://privatebin.info/directory/random?https=true
```

This will make it easy for someone to just pick a random instance.

**Comment by elrido** [comment]
Except for the redirect, this is already possible by using a tool to pick the first URL of the response (by default it returns 10 instances, shuffled):

```
curl -s https://privatebin.info/directory/api?https=true | jq -r '.[0].url'
```

In a browser context, theoretically, a [redirect](https://developer.mozilla.org/en-US/docs/Web/API/Window/location) could be done as follows:

```
const response = await fetch('https://privatebin.info/directory/api?https=true');
window.location.href = await response.json()[0].url;
```
That won't work ATM, though, as we don't set the CORS header `Access-Control-Allow-Origin` for privatebin.info/directory/api. Given that I had added the API with the idea that it could be used by client side web applications, would adding that header enable the use case that you had in mind or would you have wanted to use the API in a CLI context?

**Comment by bohwaz** [comment]
No that wouldn't work, as the idea is just to provide users with a direct link in emails, so no javascript there.

**Comment by elrido** [comment]
Fair enough, so a redirecting endpoint could cover both the email use case as well as the web client one that I had in mind, without requiring exposing the API with a CORS header. I'll look into which one is simpler to implement, adding a new parameter or creating a different endpoint.

**Comment by bohwaz** [comment]
I think that a different endpoint would be better so that you can just provide a short URL like `https://privatebin.info/directory/random` which would by default redirect you to a random secure instance.

**Comment by elrido** [comment]
I've now implemented such an endpoint in 0.10.0, it is described in https://privatebin.info/directory/about#faq-forward and the source is found in:
https://github.com/PrivateBin/Directory/blob/a0c6081a8049f089d5e6b963767be67ed858a636/src/main.rs#L385-L441

**Comment by bohwaz** [comment]
Wonderful, thank you :)
